src/main/java/core/basesyntax/operations/ReturnOperation.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void apply(Storage storage, FruitTransaction transaction) {
+        if (storage == null) {
+            throw new IllegalArgumentException("Storage cannot be null.");
+        }
+        if (transaction == null) {
+            throw new IllegalArgumentException("Transaction cannot be null.");
+        }
+        if (transaction.getQuantity() <= 0) {
+            throw new IllegalArgumentException("Quantity must be greater than zero.");
+        }
+
+        int currentQuantity = storage.getQuantity(transaction.getFruit());
+        int newQuantity = currentQuantity + transaction.getQuantity();
+
+        storage.updateFruit(transaction.getFruit(), newQuantity);

```

Comment:
The `apply` method should not directly interact with the `Storage` class. Instead, consider following the Dependency Inversion Principle (DIP) by depending on abstractions rather than concrete implementations. You could define an interface for the storage operations and have the `Storage` class implement this interface. This would make the code more flexible and easier to test.